src/halmos/sevm.py

```diff
@@ -2049,38 +2106,37 @@ def run(self, ex0: Exec) -> Tuple[List[Exec], Steps, HalmosLogs]:
 
                 if self.options.get("log"):
                     if opcode == EVM.JUMPI:
-                        steps[step_id] = {"parent": prev_step_id, "exec": str(ex)}
+                        self.steps[step_id] = {"parent": prev_step_id, "exec": str(ex)}
                     # elif opcode == EVM.CALL:
-                    #     steps[step_id] = {'parent': prev_step_id, 'exec': str(ex) + ex.st.str_memory() + '\n'}
+                    #     self.steps[step_id] = {'parent': prev_step_id, 'exec': str(ex) + ex.st.str_memory() + '\n'}
                     else:
-                        # steps[step_id] = {'parent': prev_step_id, 'exec': ex.summary()}
-                        steps[step_id] = {"parent": prev_step_id, "exec": str(ex)}
+                        # self.steps[step_id] = {'parent': prev_step_id, 'exec': ex.summary()}
+                        self.steps[step_id] = {"parent": prev_step_id, "exec": str(ex)}
```

Comment:
not directly related to this PR, but it looks like both branches are doing the same thing and could be simplified

Reply:
good catch! it's legacy code, and i need to clean this up anyway; will do this later.